src/editor/editor.ts

```diff
@@ -11,21 +11,30 @@ import {jsonAttr} from '@exadel/esl/modules/esl-base-element/core';
 import {UIPPlugin} from '../core/plugin';
 
 interface EditorConfig {
-  theme: string;
-  mode: string;
-  printMarginColumn: number;
-  wrap: number;
+  theme?: string;
+  mode?: string;
+  printMarginColumn?: number;
+  wrap?: number | boolean;
+}
+
+interface Theme {
+  [index: string]: string;
 }
 
 export class UIPEditor extends UIPPlugin {
   public static is = 'uip-editor';
-  public static defaultOptions = {
+  public static defaultOptions: EditorConfig = {
     theme: 'ace/theme/chrome',
     mode: 'ace/mode/html',
     printMarginColumn: -1,
     wrap: true,
   };
 
+  static themes: Theme = {
```

Comment:
Propose to rename to something like `themesMapping`
Use UPPERCASE if it's more like a final system mapping (constant).
Usual camelCase like it is right now in case we do not restrict for the user to extend or change it.